### PR TITLE
[SYCL][NFC] Refactor RT unit tests

### PR DESCRIPTION
**Pull request by romanovvlad**
Merged get_native_interop and spec_constants tests into SYCL2020 test suite,
introduced a header with common PI API redefinitions.

### CODE REVIEW

**Review by bader — COMMENTED** [review]
LGTM.
A couple of minor comments.

**Review by bader — APPROVED** [review]
Thanks!

File: sycl/unittests/SYCL2020/GetNativeOpenCL.cpp [hunk 1]
```diff
@@ -13,9 +11,12 @@
...
 #include <detail/context_impl.hpp>
-#include <gtest/gtest.h>
+
+#include <helpers/CommonRedefinitions.hpp>
 #include <helpers/PiMock.hpp>
 
+#include <gtest/gtest.h>
+
 #include <iostream>
 #include <memory>
```
Comment (alexbatashev) [inline sycl/unittests/SYCL2020/GetNativeOpenCL.cpp:21]: For `sycl::program` usages.

```suggestion
#include <memory>

#define SYCL2020_DISABLE_DEPRECATION_WARNINGS
```

File: sycl/unittests/helpers/CommonRedefinitions.hpp [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+#include <CL/sycl.hpp>
+#include <helpers/PiImage.hpp>
+#include <helpers/PiMock.hpp>
+
+inline pi_result redefinedProgramCreateCommon(pi_context, const void *, size_t,
+                                              pi_program *) {
+  return PI_SUCCESS;
+}
+
```
Comment (alexbatashev) [inline sycl/unittests/helpers/CommonRedefinitions.hpp:17]: Nit. I would allocate something like `int *` and release it just in case we test somewhere that the program is not null. This can be applied as a separate PR, though.

**Review by alexbatashev — APPROVED** [review]
LGTM

File: sycl/unittests/SYCL2020/GetNativeOpenCL.cpp (hunk 1 above)
Comment (bader) [reply sycl/unittests/SYCL2020/GetNativeOpenCL.cpp:21]: This is SYCL2020 test, so there must be no `sycl::program` usages. Right?
I suggest we do not use deprecated API in our unittests and have dedicated set of tests to validate that deprecated API is supported until we decide to drop such support. Does it make sense?

Comment (alexbatashev) [reply sycl/unittests/SYCL2020/GetNativeOpenCL.cpp:21]: It does

Comment (bader) [reply sycl/unittests/SYCL2020/GetNativeOpenCL.cpp:21]: Let's address both your comments in a separate PR, if that's okay.
